src/transformers/models/glm/modeling_glm.py

```diff
@@ -698,14 +690,16 @@ def __init__(self, config: GlmConfig):
         super().__init__(config)
         self.padding_idx = config.pad_token_id
         self.vocab_size = config.vocab_size
-
         self.embed_tokens = nn.Embedding(config.vocab_size, config.hidden_size, self.padding_idx)
         self.layers = nn.ModuleList(
             [GlmDecoderLayer(config, layer_idx) for layer_idx in range(config.num_hidden_layers)]
         )
         self.norm = GlmRMSNorm(config.hidden_size, eps=config.rms_norm_eps)
+        self.partial_rotary_factor = config.partial_rotary_factor
         self.rotary_emb = GlmRotaryEmbedding(
-            dim=config.head_dim // 2, max_position_embeddings=config.max_position_embeddings, base=config.rope_theta
+            dim=config.head_dim * self.partial_rotary_factor,
+            max_position_embeddings=config.max_position_embeddings,
+            base=config.rope_theta,
```

Comment:
Yes, that's indeed the only change that should be applied to the RotaryEmbedding 🤗